### PR TITLE
Enable mypy disallow_untyped_defs on raiden.transfer module

### PR DESCRIPTION
**Pull request by palango**
Enable the check, otherwise we will regress.

### CODE REVIEW

File: raiden/transfer/views.py [hunk 1]
```diff
@@ -72,19 +77,21 @@ def count_token_network_channels(
     return count
 
 
-def state_from_raiden(raiden) -> ChainState:  # pragma: no unittest
-    return raiden.wal.state_manager.current_state
+def state_from_raiden(raiden: "RaidenService") -> ChainState:  # pragma: no unittest
+    assert raiden.wal, "raiden.wal not set"
+    # TODO: current_state should not be optional
+    return raiden.wal.state_manager.current_state  # type: ignore
```
Comment (palango) [inline raiden/transfer/views.py]: I added this ignore here for now, until we change `current_state` to be non-optional.

Comment (karlb) [reply raiden/transfer/views.py]: You could use an assert instead.

Comment (palango) [reply raiden/transfer/views.py]: I had that before, but then the `smoketest` fails.

Comment (palango) [reply raiden/transfer/views.py]:
```
685 Traceback (most recent call last):
686   File "/Users/paul/Projects/brainbot/raiden/raiden/ui/cli.py", line 677, in smoketest
687     token=token,
688   File "/Users/paul/Projects/brainbot/raiden/raiden/tests/utils/smoketest.py", line 374, in run_smo    ketest
689     app = run_app(**args)                                                                          690   File "/Users/paul/Projects/brainbot/raiden/raiden/ui/app.py", line 290, in run_app
691     raiden_app.start()
692   File "/Users/paul/Projects/brainbot/raiden/raiden/app.py", line 127, in start
693     self.raiden.start()
694   File "/Users/paul/Projects/brainbot/raiden/raiden/raiden_service.py", line 386, in start
695     self.handle_and_track_state_changes([init_state_change, new_network_state_change])
696   File "/Users/paul/Projects/brainbot/raiden/raiden/raiden_service.py", line 573, in handle_and_tra    ck_state_changes
697     for greenlet in self.handle_state_changes(state_changes):
698   File "/Users/paul/Projects/brainbot/raiden/raiden/raiden_service.py", line 592, in handle_state_c    hanges                                                                                             699     old_state = views.state_from_raiden(self)
700   File "/Users/paul/Projects/brainbot/raiden/raiden/transfer/views.py", line 82, in state_from_raid    en
701     assert raiden.wal.state_manager.current_state, "raiden.wal.state_manager.current_state not set"
702 AssertionError: raiden.wal.state_manager.current_state not set

Comment (karlb) [reply raiden/transfer/views.py]: Ah, that was the part we it actually is None during the setup. Then let's go with the ignore.

**Pull request merged**